tests/runtime/servicers/test_training_management_servicer.py

```diff
@@ -75,6 +75,72 @@ def test_training_runs(training_management_servicer, training_pool):
     assert response.status == TrainingStatus.COMPLETED.value
 
 
+def test_training_can_cancel(training_management_servicer):
+    # Create a future and set it in the training manager
+    event = threading.Event()
+    model_future = MODEL_MANAGER.train(
+        SampleModule,
+        DataStream.from_iterable([]),
+        wait_event=event,
+    )
+
+    # send a request, check it's not errored
+    request = TrainingInfoRequest(training_id=model_future.id).to_proto()
+    response = training_management_servicer.GetTrainingStatus(request, context=None)
+    assert response.status != TrainingStatus.ERRORED.value
+
+    event.set()
+    model_future.wait()
+
+    # cancel the training request, check that its status is now canceled
+    training_management_servicer.CancelTraining(request, context=None)
```

Comment:
Interesting. I would expect that calling cancel on the service _after_ calling `model_future.wait()` would show that the model has completed. This may be a bug in the status reporting for the local trainer.

Reply:
I think what this is doing (I copied the `wait_event` and `model_future.wait()` from another test) is the `wait_event` is telling the model's train [function](https://github.com/caikit/caikit/blob/main/tests/fixtures/sample_lib/modules/sample_task/sample_implementation.py#L125) "wait, don't complete the train". And then the model_future.wait() is waiting for that training to complete and therefore will not reach "COMPLETED" state? So I can do CancelTraining here while it's still in state RUNNING.

Do you expect the `model_future.wait()` to make it `COMPLETED`?